server/service/service.go

```diff
@@ -384,7 +437,42 @@ func (s *Service) BootstrapStream(stream bpb.Bootstrap_BootstrapStreamServer) er
 			log.Infof("Successfully sent bootstrap data to %s", deviceID)
 
 		case *bpb.BootstrapStreamRequest_ReportStatusRequest:
-			// TODO: process request
+			log.Infof("=============================================================================")
+			log.Infof("====================== Stream status report received ======================")
+			log.Infof("=============================================================================")
+			log.Infof("Received ReportStatusRequest from %s: %+v", deviceID, req.ReportStatusRequest)
+
+			if session.currentState == stateInitial {
+				log.Info("Received ReportStatusRequest on a new stream. Starting re-authentication...")
+				lu, ccSerial, err := buildLookupFromReportStatus(ctx, req.ReportStatusRequest)
+				if err != nil {
+					return err
+				}
+
+				newSession, newDeviceID, err := s.establishSessionAndSendChallenge(stream, ctx, lu, req.ReportStatusRequest.GetIdentity(), ccSerial)
+				if err != nil {
+					return err
+				}
+				newSession.pendingStatusReport = req.ReportStatusRequest
```

Comment:
What's the need for pendingStatusReport?

Reply:
sorry that was a left over code, removed